.github/copilot-instructions.md

```diff
@@ -0,0 +1,168 @@
+  - YAML (CI/CD pipelines, Porter manifests)
+
+- **Cloud Services**:
+  - Azure services (App Service, Container Registry, Cosmos DB, etc.)
+  - Microsoft Entra ID for authentication
+  - Azure VMSS for resource processor
+
+- **Development Tools**:
+  - Docker for containerization
+  - GitHub Actions for CI/CD
+  - Make for build/deployment automation
+
+## Repository Structure
+
+```text
+├── .github               - GitHub workflows, issue templates, and configuration
+├── devops                - DevOps scripts and bootstrapping tools
+├── docs                  - Documentation
+├── e2e_tests             - pytest-based end-to-end tests
+├── api_app               - API source code and docs
+├── resource_processor    - VMSS Porter Runner
+├── scripts               - Utility scripts
+└── templates             - Resource templates
+    ├── core/terraform    - Terraform definitions of Azure TRE core resources
+    ├── shared_services   - Terraform definitions of shared services
+    ├── workspace_services - Workspace services
+    └── workspaces        - Workspace templates
+```
+
+## Coding Conventions
+
+- **Python**:
+  - Follow PEP 8 style guidelines
+  - Use FastAPI for API endpoints
+  - Use pytest for testing
+
+- **Terraform**:
+  - Use HCL format
+  - Follow module structure conventions
+  - Use variables.tf and outputs.tf for module interfaces
+  - Include resource tagging for cost tracking
+
+- **TypeScript/JavaScript**:
+  - Follow standard ESLint configuration
+
+- **YAML**:
+  - Use consistent indentation (2 spaces)
+  - Follow Porter best practices for bundle manifests
+
+- **Git**:
+  - Update CHANGELOG.md for all significant changes
+  - Reference issue numbers in commit messages
+
+## Environment Assumptions
+
+- **Azure**: All resources are deployed in Azure
+- **Microsoft Entra ID**: Used for authentication and authorization
+- **Networking**: Core infrastructure uses hub-spoke networking model
+- **Security**: Zero-trust security model with strict network boundaries
+- **Deployment**: CI/CD through GitHub Actions
+
+## Template Structure
+
+Azure TRE uses Porter bundles to define workspaces, workspace services, and user resources. These bundles consist of:
+
+### porter.yaml
+
+This is the main Porter manifest file that defines:
+- Credentials required for deployment
+- Parameters and their defaults
+- Actions (install, upgrade, uninstall)
+- Mixins used (terraform, exec, etc.)
+- Outputs from the deployment
+
+Example structure:
+```yaml
+name: tre-service-example
+version: 0.1.0
+description: "An example TRE service"
+registry: azuretre
+...
+```
+
+### template_schema.json
+
+JSON Schema file that defines the parameters that can be provided when deploying a resource. It follows standard JSON Schema format and is used by the API and UI to generate forms for resource creation.
+
+Key sections include:
+- Properties with types, descriptions, and defaults
+- Required fields
+- UI schema for customizing the display in the UI
+
```

Comment:
the properties names in template_schema.json and the parameters section in porter.yaml should match